### PR TITLE
clippy warnings

### PR DESCRIPTION
**Pull request by NegrilaRares**
ran --fix with clippy 

after some tests if everything is in order ill solve the remaining ones i can by hand

### CODE REVIEW

**Comment by NegrilaRares** [comment]
found the fix for previous problem it would seem i had a misunderstanding

File: tock-process-console/src/board/connection.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 // Copyright OXIDOS AUTOMOTIVE 2024.
 
+#![warn(clippy::new_ret_no_self)]
```
Comment (george-cosma) [inline tock-process-console/src/board/connection.rs:5]: I don't think this has any use

File: tock-process-console/src/ui_management/components/component.rs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 // Copyright OXIDOS AUTOMOTIVE 2024.
 
+#![warn(clippy::new_ret_no_self)]
```
Comment (george-cosma) [inline tock-process-console/src/ui_management/components/component.rs:5]: I don't think this has any use

File: tock-process-console/src/ui_management/pages/mod.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Licensed under the Apache License, Version 2.0 or the MIT License.
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 // Copyright OXIDOS AUTOMOTIVE 2024.
-
+#[allow(unused)]
```
Comment (george-cosma) [inline tock-process-console/src/ui_management/pages/mod.rs:4]: I don't think this has any use

File: tock-process-console/src/ui_management/pages/mod.rs [hunk 4]
```diff
@@ -62,10 +55,6 @@ impl Component for AppRouter {
         }
     }
 
-    fn name(&self) -> &str {
```
Comment (george-cosma) [inline tock-process-console/src/ui_management/pages/mod.rs:65]: Didn't we discuss that this was an error from clippy?

File: tock-process-console/src/ui_management/pages/terminal_page/applications_page.rs [hunk 5]
```diff
@@ -2,14 +2,15 @@
...
         apps_list::{self, AppsList},
         terminal_box::{RenderProps, TerminalBox},
     },
     section::{
         self,
-        usage::{HasUsageInfo, UsageInfo, UsageInfoLine},
+        usage::{UsageInfo, UsageInfoLine},
         SectionActivation,
     },
 };
```
Comment (george-cosma) [inline tock-process-console/src/ui_management/pages/terminal_page/applications_page.rs:16]: We can fix this issue, we don't need to allow it
```rust
use super::{
    components::{
        apps_list::{self, AppsList},
        terminal_box::{RenderProps, TerminalBox},
    },
    section::SectionActivation,
};
```

File: tock-process-console/src/ui_management/pages/terminal_page/applications_page.rs [hunk 6]
```diff
@@ -345,38 +330,3 @@ impl ComponentRender<apps_list::RenderProperties> for ApplicationsPage {
         // frame.render_widget(usage, right);
     }
 }
-
```
Comment (george-cosma) [inline tock-process-console/src/ui_management/pages/terminal_page/applications_page.rs:348]: Why did you remove tihs? Unused?

File: tockloader/src/board_settings.rs [hunk 7]
```diff
@@ -1,3 +1,5 @@
+#[allow(dead_code)]
```
Comment (george-cosma) [inline tockloader/src/board_settings.rs:1]: This will be fixed on Ami's branch, I would mobe it to the `board` field rather than the whole struct

**Review by george-cosma — CHANGES_REQUESTED** [review]

**Review by george-cosma — APPROVED** [review]
